src/orion/service/client/workon.py

```diff
@@ -0,0 +1,178 @@
+    RemoteException,
+    RemoteExperiment,
+    RemoteTrial,
+)
+
+log = logging.getLogger(__name__)
+
+
+class WorkonClientREST(BaseClientREST):
+    """Minimal REST API, only implements the required method to run workon.
+
+    Its goal is limited to communicating to the remote algo,
+    For generic query you should implement the functionality inside the rest storage.
+
+    """
+
+    def __init__(self, endpoint, token) -> None:
+        super().__init__(endpoint, token)
+        self.experiment = None
+        self._pacemakers = {}
+
+    @property
+    def experiment_name(self) -> Optional[str]:
+        """returns the current experiment name"""
+        return self.experiment.name if self.experiment else None
+
+    @property
+    def experiment_id(self) -> Optional[str]:
+        """returns the current experiment id"""
+        return self.experiment.euid if self.experiment else None
+
+    def new_experiment(self, name, **config) -> RemoteException:
+        """Create a new experiment"""
+        payload = self._post("experiment", name=name, **config)
+
+        payload.pop("experiment_name", None)
+        payload["name"] = name
+        self.experiment = RemoteExperiment(**payload)
+        return self.experiment
+
+    def suggest(self, pool_size: int = 1, experiment_name=None) -> TrialCM:
+        """Generate a new trial for the current experiment"""
+        experiment_name = experiment_name or self.experiment_name
+
+        if experiment_name is None:
+            raise ExperiementIsNotSetup("experiment_name is not set")
+
+        log.debug("client: suggest: %s", experiment_name)
+        result = self._post(
+            "suggest", experiment_name=experiment_name, pool_size=pool_size
+        )
+
+        trials = []
+        for trial in result["trials"]:
+            trials.append(
+                RemoteTrial(**trial, exp_working_dir=self.experiment.working_dir)
+            )
+
+        # Currently we only return a single trial
+        trial = trials[0]
+
+        # Create pacemaker for them
+        self._maintain_reservation(trial)
+        return TrialCM(self, trial)
+
+    def _maintain_reservation(self, trial):
+        self._pacemakers[trial.id] = TrialPacemaker(trial, self)
+        self._pacemakers[trial.id].start()
+
+    def release(self, trial, status="interrupted"):
+        """Release a trial after work on it finished"""
+        experiment_name = self.experiment_name
+
+        if experiment_name is None:
+            raise ExperiementIsNotSetup("experiment_name is not set")
+
+        failed = False
+        try:
+            self._post(
+                "release",
+                euid=self.experiment_id,
+                experiment_name=experiment_name,
+                trial_hash=trial.id,
+                status=status,
+            )
+        except Exception as e:
+            failed = True
+            raise e
+        finally:
+            pacemaker = self._pacemakers.pop(trial.id, None)
```

Comment:
Looks like this class could have something like `ExperimentClient._release_reservation`. Similar code is used in `observe` below.